sdk/src/settings/builder.rs

```diff
@@ -118,55 +121,44 @@ impl SettingsValidate for ThumbnailSettings {
 }
 
 /// Settings for the auto actions (e.g. created, opened, placed).
-#[allow(unused)]
+#[cfg_attr(feature = "json_schema", derive(schemars::JsonSchema))]
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
-pub(crate) struct AutoActionSettings {
+pub struct AutoActionSettings {
     /// Whether to enable this auto action or not.
     pub enabled: bool,
     /// The default source type for the auto action.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub source_type: Option<DigitalSourceType>,
 }
 
-/// Settings for how to specify the claim generator info's operating system.
-#[allow(unused)]
+#[cfg_attr(feature = "json_schema", derive(schemars::JsonSchema))]
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
-pub(crate) struct ClaimGeneratorInfoOSSettings {
-    /// Whether or not to infer the operating system.
-    pub infer: bool,
+#[serde(untagged, rename_all = "lowercase")]
+pub enum ClaimGeneratorInfoOperatingSystem {
```

Comment:
do we still need this if we are using the string "auto"?

Reply:
I think it's a good idea to express all the possible cases as Rust types. If this is constructed manually it forces you to make an informed decision. It also maps better to the JSON schemas so they can document each possible variant, with the added benefit that it provides more expressive types for the other languages.